File: Services/Catalog/src/Catalog.Domain/Entities/Product.cs
using Catalog.Domain.Enums;

namespace Catalog.Domain.Entities;

public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string ImageUrl { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = new Category();
    public int Stock { get; set; }
    public ProductUnit Unit { get; set; }
    public bool InStock { get; set; }

}

File: Frontend/vue-app/src/types/Product.ts
/**
 * Unités de mesure disponibles pour les produits
 */
export enum ProductUnit {
  PIECE = 'piece',
  KILOGRAM = 'kg',
  GRAM = 'g',
  LITER = 'L',
  MILLILITER = 'mL',
  PACK = 'pack',
  BOX = 'boîte'
}

export interface Product {
  /**
   * Identifiant unique du produit (GUID)
   */
  id: string;

  /**
   * Nom du produit
   */
  name: string;

  /**
   * Description détaillée du produit
   */
  description: string;

  /**
   * Prix du produit en euros
   */
  price: number;

  /**
   * URL de l'image du produit
   * Si non définie, une image par défaut sera utilisée
   */
  image?: string;

  /**
   * Identifiant de la catégorie (GUID)
   * Référence à Category.id
   */
  category: string;

  /**
   * Quantité en stock
   */
  stock: number;

  /**
   * Unité de mesure du produit
   */
  unit: ProductUnit;

  /**
   * Indique si le produit est disponible à la vente
   */
  inStock: boolean;

  /**
   * Date de création du produit
   */
  createdAt?: Date;

  /**
   * Date de dernière modification du produit
   */
  updatedAt?: Date;
} 